packages/spiderable/spiderable_client.js

```diff
@@ -17,14 +17,52 @@
 
 Spiderable._initialSubscriptionsStarted = false;
 
+Spiderable._onReadyHook = new Hook({
+  debugPrintExceptions: "onReadyHook callback"
+});
+
+// register a new onReady hook for validation
+Spiderable.onReady = function (fn) {
```

Comment:
I think this name is confusing; it sounds more like "callbacks to call when things become ready".  Maybe `Spiderable.addReadyCondition`?